src/types.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export type Metrics = {
+  Timestamp?: number;
+  Documents?: number;
+  Frames?: number;
+  JSEventListeners?: number;
+  Nodes?: number;
+  LayoutCount?: number;
+  RecalcStyleCount?: number;
+  LayoutDuration?: number;
+  RecalcStyleDuration?: number;
+  ScriptDuration?: number;
+  TaskDuration?: number;
+  JSHeapUsedSize?: number;
+  JSHeapTotalSize?: number;
+}
+
+export type WaitForOptions = {
+  timeout?: number;
+  waitUntil?: PuppeteerLifeCycleEvent | PuppeteerLifeCycleEvent[];
+}
+
+export type MediaFeature = {
+  name: string;
+  value: string;
+}
+
+export type ScreenshotClip = {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+export type ScreenshotOptions = {
+  type?: 'png' | 'jpeg';
+  path?: string;
+  fullPage?: boolean;
+  clip?: ScreenshotClip;
+  quality?: number;
+  omitBackground?: boolean;
+  encoding?: string;
+}
+
+type PDFMargin = {
+  top?: string | number;
+  bottom?: string | number;
+  left?: string | number;
+  right?: string | number;
+}
+
+export type PDFOptions = {
+  scale?: number;
+  displayHeaderFooter?: boolean;
+  headerTemplate?: string;
+  footerTemplate?: string;
+  printBackground?: boolean;
+  landscape?: boolean;
+  pageRanges?: string;
+  format?: string;
+  width?: string | number;
+  height?: string | number;
+  preferCSSPageSize?: boolean;
+  margin?: PDFMargin;
+  path?: string;
+}
+
+export type PaperFormat = {
+  width: number;
+  height: number;
+}
+
+export type PuppeteerLifeCycleEvent =
+  | 'load'
+  | 'domcontentloaded'
+  | 'networkidle0'
+  | 'networkidle2';
+export type ProtocolLifeCycleEvent =
+  | 'load'
+  | 'DOMContentLoaded'
+  | 'networkIdle'
+  | 'networkAlmostIdle';
```

Comment:
@christian-bromann do you have a reason for types over interfaces? I tend to lean towards interfaces but don't have a hugely strong opinion. I think we should be consistent though.

I'd also rather not create a generic `types.ts` file but instead keep these types in the modules that make sense to own them. E.g. `PuppeteerLifeCycleEvent` sits nicely inside the `LifecycleWatcher`.

There are definitely some that should be in a generic `types.ts` file but I think for the ones that have a natural home inside another module we could leave them there for now.

Reply:
> do you have a reason for types over interfaces? I tend to lean towards interfaces but don't have a hugely strong opinion. I think we should be consistent though.

Not really. I thought if we would want to collect some generic types in an own module it would make sense to actually define them as types. Let's keep things as they are for now and let me revert this specific change.